internal/provider/resources/variable.go

```diff
@@ -133,41 +136,42 @@ func (r *VariableResource) Schema(_ context.Context, _ resource.SchemaRequest, r
 	}
 }
 
-// copyVariableToModel copies an api.Variable to a VariableResourceModel.
-func copyVariableToModel(ctx context.Context, variable *api.Variable, model *VariableResourceModel) diag.Diagnostics {
-	model.ID = types.StringValue(variable.ID.String())
-	model.Created = customtypes.NewTimestampPointerValue(variable.Created)
-	model.Updated = customtypes.NewTimestampPointerValue(variable.Updated)
+// copyVariableToModel maps an API response to a model that is saved in Terraform state.
+// A model can be a Terraform Plan, State, or Config object.
+func copyVariableToModel(ctx context.Context, variable *api.Variable, tfModel *VariableResourceModel) diag.Diagnostics {
+	tfModel.ID = types.StringValue(variable.ID.String())
+	tfModel.Created = customtypes.NewTimestampPointerValue(variable.Created)
+	tfModel.Updated = customtypes.NewTimestampPointerValue(variable.Updated)
 
-	model.Name = types.StringValue(variable.Name)
-	model.Value = types.StringValue(variable.Value)
+	tfModel.Name = types.StringValue(variable.Name)
+	tfModel.Value = types.StringValue(variable.Value)
 
 	tags, diags := types.ListValueFrom(ctx, types.StringType, variable.Tags)
 	if diags.HasError() {
 		return diags
 	}
-	model.Tags = tags
+	tfModel.Tags = tags
 
 	return nil
 }
 
 // Create creates the resource and sets the initial Terraform state.
 func (r *VariableResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
-	var model VariableResourceModel
+	var config VariableResourceModel
```

Comment:
I don't mind using `config` at all, I just noticed that the [Terraform provider tutorial for Create](https://developer.hashicorp.com/terraform/tutorials/providers-plugin-framework/providers-plugin-framework-resource-create#implement-create-functionality) uses `plan` here. Which do you prefer?

Reply:
it's a good question - im actually inclined to move these over to a `req.Plan` on the creates, since `req.Config` appears to be the raw, user-inputted object

```golang
// CreateRequest represents a request for the provider to create a
// resource. An instance of this request struct is supplied as an argument to
// the resource's Create function.
type CreateRequest struct {
	// Config is the configuration the user supplied for the resource.
	//
	// This configuration may contain unknown values if a user uses
	// interpolation or other functionality that would prevent Terraform
	// from knowing the value at request time.
	Config tfsdk.Config

	// Plan is the planned state for the resource.
	Plan tfsdk.Plan

	// ProviderMeta is metadata from the provider_meta block of the module.
	ProviderMeta tfsdk.Config
}
```

Reply:
updated here: https://github.com/PrefectHQ/terraform-provider-prefect/pull/199/commits/7b8a1e1641f0785b38ab4619a2cb48daaf3eb929